### PR TITLE
record: add record statistics on the public detailed view

### PR DESCRIPTION
**Pull request by jma**
* Adds a statistics property to the documents api.
* Adds the number of document view and file downloads on the documents detailed view.

Co-Authored-by: Johnny Mariéthoz <Johnny.Mariethoz@rero.ch>
Co-Authored-by: Valeria Granata <valeria@chaw.com>

## Why are you opening this PR?
Issue: https://github.com/rero/sonar/issues/400

This PR depends on:
https://github.com/rero/sonar-ui/pull/316

## Code review check list

- [ ] Commit message template compliance.
- [ ] Commit message without typos.
- [ ] File names.
- [ ] Functions names.
- [ ] Functions docstrings.
- [ ] Unnecessary commited files?
- [ ] Extracted translations?

### CODE REVIEW

File: sonar/modules/documents/templates/documents/record.html [hunk 1]
```diff
@@ -432,13 +437,28 @@ <h5 id="other_files" class="mt-5">{{ _('Other files') }}</h5>
...
     {% endfor %}
   </div>
   {% endif %}
 
+  <!-- Statistics -->
+  <h5 id="stats" class="mt-5">{{ _('Statistics') }}</h5>
+  <hr class="mb-4 mt-0" />
+  <div class="row">
+    <div class="col">
+      <strong>{{ _('Document views') }}</strong>: {{stats['record-view']}}
```
Comment (vgranata) [inline sonar/modules/documents/templates/documents/record.html:452]: Make semi-column bold.
`<strong>{{ _('Document views') }}:</strong> {{stats['record-view']}}`

**Review by vgranata — APPROVED** [review]

File: tests/ui/documents/test_documents_api.py [hunk 2]
```diff
@@ -141,3 +144,38 @@ def test_is_open_access(document, embargo_date):
...
+                pid_value=document_with_file['pid'])).status_code == 200
+
+    assert client.get(
+        url_for(
+            'invenio_records_ui.doc_files',
+            pid_value=document_with_file['pid'],
+            filename='test1.pdf')
+    ).status_code == 200
+
+    # assert client.get(
```
Comment (BadrAly) [inline tests/ui/documents/test_documents_api.py:165]: remove the commented code

**Review by BadrAly — APPROVED** [review]

File: sonar/modules/documents/templates/documents/record.html [hunk 3]
```diff
@@ -432,13 +437,28 @@ <h5 id="other_files" class="mt-5">{{ _('Other files') }}</h5>
...
+  <!-- Statistics -->
+  <h5 id="stats" class="mt-5">{{ _('Statistics') }}</h5>
+  <hr class="mb-4 mt-0" />
+  <div class="row">
+    <div class="col">
+      <strong>{{ _('Document views') }}:</strong> {{stats['record-view']}}
+      <strong class="mt-2 d-block">{{ _('File downloads') }}:</strong>
+      <ul>
+        {% for key, value in stats['file-download'].items() %}
+        <li>{{key}}: {{value}}</li>
```
Comment (Garfield-fr) [inline sonar/modules/documents/templates/documents/record.html:456]: Space after `{` and before `}`

File: sonar/modules/documents/templates/documents/record.html [hunk 4]
```diff
@@ -432,13 +437,28 @@ <h5 id="other_files" class="mt-5">{{ _('Other files') }}</h5>
...
     {% endfor %}
   </div>
   {% endif %}
 
+  <!-- Statistics -->
+  <h5 id="stats" class="mt-5">{{ _('Statistics') }}</h5>
+  <hr class="mb-4 mt-0" />
+  <div class="row">
+    <div class="col">
+      <strong>{{ _('Document views') }}:</strong> {{stats['record-view']}}
```
Comment (Garfield-fr) [inline sonar/modules/documents/templates/documents/record.html:452]: Space after { and before }

File: sonar/theme/templates/sonar/macros/macro.html [hunk 5]
```diff
@@ -49,6 +49,9 @@
       <i class="fa fa-download mx-1"></i>
     </a>
     {% endif %}
+    <a href="{{ request.url }}#stats" class="sonar-tooltip" data-toggle="tooltip" data-placement="top" title="Download: {{n_download}}">
```
Comment (Garfield-fr) [inline sonar/theme/templates/sonar/macros/macro.html:52]: Space after { and before }

**Review by Garfield-fr — APPROVED** [review]

**Comment by pronguen** [comment]
- [ ] *... (waiting 1 day to see the increment of the stat)*
- Ideally we should display the file labels instead of the file names in the statistics section. This can be done later, as I just opened an issue: https://github.com/rero/sonar/issues/882

**Pull request merged**